homeassistant/components/qnap_qsw/sensor.py

```diff
@@ -116,7 +187,13 @@ def __init__(
     @callback
     def _async_update_attrs(self) -> None:
         """Update sensor attributes."""
-        self._attr_native_value = self.get_device_value(
+        value = self.get_device_value(
             self.entity_description.key, self.entity_description.subkey
         )
+        if value is not None:
+            if self.entity_description.subkey in [QSD_RX_OCTETS, QSD_TX_OCTETS]:
+                value = round(value / (1024**2), 2)
```

Comment:
We shouldn’t mutate the data here.  We should display it in the format that the library provides.  Changing the unit to something else is a front end concern.

Reply:
> We shouldn’t mutate the data here. We should display it in the format that the library provides. Changing the unit to something else is a front end concern.

Done in b72de801e19b78743db3bc31e80aa3823170ac36